packages/services/src/view-service/fees.ts

```diff
@@ -117,9 +118,12 @@ export const getAssetFromGasPriceTable = async (
   // If a specific asset ID is provided, extracted from the transaction request, check its balance is
   // positive and GasPrices for that asset exist.
   if (assetId) {
-    const balance = await indexedDb.hasTokenBalance(request.source, assetId);
+    const balance = await indexedDb.accountHasSpendableAsset(request.source, assetId);
     // This check ensures that the alternative fee token is a valid fee token, for example, TestUSD is not.
-    const isInGasTable = altGasPrices.find(gp => gp.assetId?.equals(assetId));
```

Comment:
this checks if the `assetId` passed in exists in the `GasPrices` table. since we're comparing assetId objects directly, why isn't the equality method sufficient here? can you explain when we should use Bech32m encoded string representations instead?